### PR TITLE
Canny performance part1

### PR DESCRIPTION
**Pull request by ibogosavljevic**
Add source code for canny performance part 1 blog post

### CODE REVIEW

File: canny-performance-part1/canny-run.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+CC=clang
+CFLAGS="-O3 -ffast-math -g -lm -fno-omit-frame-pointer"
+
+${CC} ${CFLAGS} canny-optimized.c -o canny-optimized
+${CC} ${CFLAGS} canny-original.c -o canny-original
+
+CANNYPARAMS="testvecs/input/15360_8640.pgm 0.5 0.7 0.9"
+TIMECMD="/usr/bin/time -f'%E'"
```
Comment (javinovo) [inline canny-performance-part1/canny-run.sh:13]:
```suggestion
CANNYPARAMS="testvecs/input/15360_8640.pgm 0.5 0.7 0.9"

TIMECMD="/usr/bin/time -f'%E'"
```

File: canny-performance-part1/canny-original.c [hunk 2]
```diff
@@ -0,0 +1,1042 @@
...
+void make_gaussian_kernel(float sigma, float **kernel, int *windowsize);
+void derrivative_x_y(short int *smoothedim, int rows, int cols,
+    short int **delta_x, short int **delta_y);
+void magnitude_x_y(short int *delta_x, short int *delta_y, int rows, int cols,
+    short int **magnitude);
+void apply_hysteresis(short int *mag, unsigned char *nms, int rows, int cols,
+    float tlow, float thigh, unsigned char *edge);
+void radian_direction(short int *delta_x, short int *delta_y, int rows,
+    int cols, float **dir_radians, int xdirtag, int ydirtag);
+double angle_radians(double x, double y);
```
Comment (javinovo) [inline canny-performance-part1/canny-original.c:36]: Are all these declarations required? I think only a subset are really needed. If you'd rather list all the declare all functions here I think there are some missing such as `canny`.

Comment (javinovo) [reply canny-performance-part1/canny-original.c:36]: Also, this is missing from `canny-optimized.c` so please make it consistent

Comment (ibogosavljevic) [reply canny-performance-part1/canny-original.c:36]: These are taken from the header file. Not sure if all of them are needed, but they don't pose a problem so I wouldn't remove them.  As far as I see, files canny-optimized.c and canny-original.c are identical except for the changes in gaussian_smooth, derivative_x_y and apply_hysteresis.

**Review by javinovo — APPROVED** [review]

**Pull request merged**